x-pack/plugins/triggers_actions_ui/public/application/components/rules_setting/query_delay/rules_settings_query_delay_section.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo } from 'react';
+import {
+  MAX_QUERY_DELAY,
+  MIN_QUERY_DELAY,
+  RulesSettingsQueryDelayProperties,
+} from '@kbn/alerting-plugin/common';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiSpacer,
+  EuiText,
+  EuiEmptyPrompt,
+  EuiTitle,
+} from '@elastic/eui';
+import { RulesSettingsRange } from '../rules_settings_range';
+
+const queryDelayDescription = i18n.translate(
+  'xpack.triggersActionsUI.rulesSettings.modal.queryDelayDescription',
+  {
+    defaultMessage:
+      'Increase the query time window to mitigate gaps in data availability due to the ES index refresh interval.',
```

Comment:
Some initial thoughts/research:

- In machine learning anomaly detection jobs, they have a query delay, which they label as "Time delay in seconds, between current time and latest input data time." However I don't think the "input data time" is as relevant in this case.
- The concept of the "query time window" is clearer in some rules (e.g. ES query rule) than others (e.g. anomaly detection rule). In the former, we have a tip for the time window fields, which say "The time window indicates how far back in time to search".
- There are docs about the [index.refresh_interval](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-refresh-interval-setting) and the [impact](https://www.elastic.co/guide/en/elasticsearch/reference/current/tune-for-indexing-speed.html#_unset_or_increase_the_refresh_interval) of increasing the interval.

With this in mind, here are some options for consideration:
> Mitigate gaps in data availability by adding a time delay to all rule queries.
> Mitigate gaps in data availability by searching farther back in time.

Or, if we have to explicitly call out the refresh interval as the culprit:
> Delay all rule queries to mitigate the impact of an increased index refresh interval on data availability.